### PR TITLE
drivers: serial: stm32: always enable clock when exiting low power

### PR DESCRIPTION
**Pull request by gautierg-st**
Systematically enable the UART clock again when exiting a low power mode before reading the UART register.
Even though the previous code worked on STM32WBA, for most series, it is necessary to enable the clock to access the registers, otherwise they read as 0.

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
I have an issue with this as initially pinctrl config was done before UART clock activation to avoid glitches on the line.
Is there any other way to do this ?

**Comment by gautierg-st** [comment]
I suppose we'll have to do the pinctrl, then the clock, and if necessary the full init (or maybe just the rest of the init).

**Review by erwango — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]
minor else LGTM

File: drivers/serial/uart_stm32.c [hunk 1]
```diff
@@ -2081,25 +2081,26 @@ static int uart_stm32_pm_action(const struct device *dev,
...
-					       (clock_control_subsys_t)&config->pclken[0]);
-			if (err != 0) {
-				LOG_ERR("Could not enable (LP)UART clock");
-				return err;
-			}
-		} else {
+		/* Enable clock */
+		err = clock_control_on(data->clock,
+					(clock_control_subsys_t)&config->pclken[0]);
+		if (err != 0) {
```
[the remark this answers is not in the document]
Comment (gautierg-st) [reply drivers/serial/uart_stm32.c]: Done

**Review by erwango — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**